pybamm/models/submodels/electrolyte_conductivity/integrated_conductivity.py

```diff
@@ -0,0 +1,173 @@
+
+        i_boundary_cc_0 = variables["Leading-order current collector current density"]
+        c_e_n = variables["Negative electrolyte concentration"]
+        c_e_s = variables["Separator electrolyte concentration"]
+        c_e_p = variables["Positive electrolyte concentration"]
+        c_e_n0 = pybamm.boundary_value(c_e_n, "left")
+
+        delta_phi_n_av = variables[
+            "X-averaged negative electrode surface potential difference"
+        ]
+        phi_s_n_av = variables["X-averaged negative electrode potential"]
+
+        tor_n = variables["Negative electrolyte tortuosity"]
+        tor_s = variables["Separator tortuosity"]
+        tor_p = variables["Positive electrolyte tortuosity"]
+
+        T_av = variables["X-averaged cell temperature"]
+        T_av_n = pybamm.PrimaryBroadcast(T_av, "negative electrode")
+        T_av_s = pybamm.PrimaryBroadcast(T_av, "separator")
+        T_av_p = pybamm.PrimaryBroadcast(T_av, "positive electrode")
+
+        param = self.param
+        l_n = param.l_n
+        l_p = param.l_p
+        x_n = pybamm.standard_spatial_vars.x_n
+        x_s = pybamm.standard_spatial_vars.x_s
+        x_p = pybamm.standard_spatial_vars.x_p
+        x_n_edge = pybamm.standard_spatial_vars.x_n_edge
+        x_p_edge = pybamm.standard_spatial_vars.x_p_edge
+
+        chi_av = param.chi(c_e_av)
+        chi_av_n = pybamm.PrimaryBroadcast(chi_av, "negative electrode")
+        chi_av_s = pybamm.PrimaryBroadcast(chi_av, "separator")
+        chi_av_p = pybamm.PrimaryBroadcast(chi_av, "positive electrode")
+
+        # electrolyte current
+        i_e_n = i_boundary_cc_0 * x_n / l_n
+        i_e_s = pybamm.PrimaryBroadcast(i_boundary_cc_0, "separator")
+        i_e_p = i_boundary_cc_0 * (1 - x_p) / l_p
+        i_e = pybamm.Concatenation(i_e_n, i_e_s, i_e_p)
+
+        i_e_n_edge = i_boundary_cc_0 * x_n_edge / l_n
+        i_e_s_edge = pybamm.PrimaryBroadcastToEdges(i_boundary_cc_0, "separator")
+        i_e_p_edge = i_boundary_cc_0 * (1 - x_p_edge) / l_p
+
+        # electrolyte potential
+        indef_integral_n = (
+            pybamm.IndefiniteIntegral(
+                i_e_n_edge / (param.kappa_e(c_e_n, T_av_n) * tor_n), x_n
+            )
+            * param.C_e
+            / param.gamma_e
+        )
+        indef_integral_s = (
+            pybamm.IndefiniteIntegral(
+                i_e_s_edge / (param.kappa_e(c_e_s, T_av_s) * tor_s), x_s
+            )
+            * param.C_e
+            / param.gamma_e
+        )
+        indef_integral_p = (
+            pybamm.IndefiniteIntegral(
+                i_e_p_edge / (param.kappa_e(c_e_p, T_av_p) * tor_p), x_p
+            )
+            * param.C_e
+            / param.gamma_e
+        )
+
+        integral_n = indef_integral_n - pybamm.boundary_value(indef_integral_n, "left")
+        integral_s = (
+            indef_integral_s
+            - pybamm.boundary_value(indef_integral_s, "left")
+            + pybamm.boundary_value(integral_n, "right")
```

Comment:
`pybamm.Integral(a)` should give the same value as `pybamm.boundary_value(pybamm.IndefiniteIntegral(a), "right")` while building a smaller tree

Reply:
but maybe it automatically gets simplified down to the same tree anyway

Reply:
How can I check if they reduce to the same tree? In any case, I think that if they are equivalent, using `pybamm.Integral(a)` and removing the boundary values on the left should result in a much more readable code.

Reply:
Just tried replacing the boundary value at the "right" by `Integral` and I get a very long error (see below) which basically looks like a dimension mismatch. So I guess they are not the same object...


```
Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 817, in process_symbol
    return self._discretised_symbols[symbol.id]
KeyError: 1041553401910644705

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 817, in process_symbol
    return self._discretised_symbols[symbol.id]
KeyError: 2459353937759019131

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 817, in process_symbol
    return self._discretised_symbols[symbol.id]
KeyError: 4130260032263747209

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 817, in process_symbol
    return self._discretised_symbols[symbol.id]
KeyError: 7060176036539492958

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 817, in process_symbol
    return self._discretised_symbols[symbol.id]
KeyError: 8911840126246680561

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 817, in process_symbol
    return self._discretised_symbols[symbol.id]
KeyError: -7919469058354380383

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/expression_tree/symbol.py", line 594, in evaluate_for_shape
    return self._saved_evaluate_for_shape
AttributeError: 'MatrixMultiplication' object has no attribute '_saved_evaluate_for_shape'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/PyBaMM/pybamm/expression_tree/symbol.py", line 801, in test_shape
    self.shape_for_testing
  File "/home/ferranbrosa/PyBaMM/pybamm/expression_tree/symbol.py", line 785, in shape_for_testing
    evaluated_self = self.evaluate_for_shape()
  File "/home/ferranbrosa/PyBaMM/pybamm/expression_tree/symbol.py", line 596, in evaluate_for_shape
    self._saved_evaluate_for_shape = self._evaluate_for_shape()
  File "/home/ferranbrosa/PyBaMM/pybamm/expression_tree/binary_operators.py", line 201, in _evaluate_for_shape
    return self._binary_evaluate(left, right)
  File "/home/ferranbrosa/PyBaMM/pybamm/expression_tree/binary_operators.py", line 497, in _binary_evaluate
    return left @ right
  File "/home/ferranbrosa/coding/reduced-TEC/env/lib/python3.7/site-packages/scipy/sparse/base.py", line 564, in __matmul__
    return self.__mul__(other)
  File "/home/ferranbrosa/coding/reduced-TEC/env/lib/python3.7/site-packages/scipy/sparse/base.py", line 502, in __mul__
    raise ValueError('dimension mismatch')
ValueError: dimension mismatch

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ferranbrosa/miniconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/ferranbrosa/miniconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/ferranbrosa/.vscode-server/extensions/ms-python.python-2020.9.114305/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/home/ferranbrosa/.vscode-server/extensions/ms-python.python-2020.9.114305/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/ferranbrosa/.vscode-server/extensions/ms-python.python-2020.9.114305/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 267, in run_file
    runpy.run_path(options.target, run_name=compat.force_str("__main__"))
  File "/home/ferranbrosa/miniconda3/lib/python3.7/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/home/ferranbrosa/miniconda3/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/home/ferranbrosa/miniconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/ferranbrosa/coding/reduced-TEC/TSPMe_compare_DFN.py", line 91, in <module>
    sim.solve([0, 3700 / Crate])
  File "/home/ferranbrosa/PyBaMM/pybamm/simulation.py", line 338, in solve
    self.build(check_model=check_model)
  File "/home/ferranbrosa/PyBaMM/pybamm/simulation.py", line 291, in build
    self._model_with_set_params, inplace=False, check_model=check_model
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 173, in process_model
    self.set_internal_boundary_conditions(model)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 419, in set_internal_boundary_conditions
    rbc = (boundary_gradient(first_orphan, next_orphan), "Neumann")
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 399, in boundary_gradient
    right_symbol_disc = self.process_symbol(right_symbol)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 819, in process_symbol
    discretised_symbol = self._process_symbol(symbol)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 853, in _process_symbol
    disc_right = self.process_symbol(right)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 819, in process_symbol
    discretised_symbol = self._process_symbol(symbol)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 853, in _process_symbol
    disc_right = self.process_symbol(right)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 819, in process_symbol
    discretised_symbol = self._process_symbol(symbol)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 862, in _process_symbol
    disc_child = self.process_symbol(child)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 819, in process_symbol
    discretised_symbol = self._process_symbol(symbol)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 852, in _process_symbol
    disc_left = self.process_symbol(left)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 819, in process_symbol
    discretised_symbol = self._process_symbol(symbol)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 852, in _process_symbol
    disc_left = self.process_symbol(left)
  File "/home/ferranbrosa/PyBaMM/pybamm/discretisations/discretisation.py", line 821, in process_symbol
    discretised_symbol.test_shape()
  File "/home/ferranbrosa/PyBaMM/pybamm/expression_tree/symbol.py", line 803, in test_shape
    raise pybamm.ShapeError("Cannot find shape (original error: {})".format(e))
```